### PR TITLE
Feat/front-end demo

### PR DESCRIPTION
**Pull request by Stefann11**

### CODE REVIEW

File: PatientWebApplication/HealthClinic.DAL/Service/RegularAppointmentService.cs [hunk 1]
```diff
@@ -268,7 +272,12 @@ private List<DoctorAppointment> CreateListOfAvailableAppointments(Shift doctorSh
             List<DoctorAppointment> availableAppointments = new List<DoctorAppointment>();
             TimeSpan time = TimeSpan.Parse(doctorShift.startTime);
             while (time != TimeSpan.Parse(doctorShift.endTime))
-            {
```
Comment (velicko-tam) [inline PatientWebApplication/HealthClinic.DAL/Service/RegularAppointmentService.cs:271]: Refactor method according to clean code principles.

**Review by velicko-tam — CHANGES_REQUESTED** [review]
PR is not approved. Marked parts need to be refactored.

File: PatientWebApplication/HealthClinic.DAL/Service/RegularAppointmentService.cs [hunk 2]
```diff
@@ -204,8 +204,12 @@ private List<DoctorAppointment> GetAvailableForFreeTime(TimeSpan time, List<Doct
         {
             while (time != TimeSpan.Parse(doctorShift.endTime) && availableAppointments.Count < 5)
             {
+                if (time < DateTime.Now.TimeOfDay && UtilityMethods.CheckIfDateIsToday(date))
+                {
+                    time = time.Add(TimeSpan.FromMinutes(15));
+                    continue;
+                }
                 availableAppointments = GetListAvailableAppointments(availableAppointments, doctorId, time, date, patientId, startTimesAppointments, operations);
```
Comment (velicko-tam) [inline PatientWebApplication/HealthClinic.DAL/Service/RegularAppointmentService.cs:212]: Refactor method so it has fewer parameters.

**Review by velicko-tam — CHANGES_REQUESTED** [review]
PR is not approved. Marked parts need to be refactored.

**Review by velicko-tam — APPROVED** [review]
PR is approved. Clean code principles were implemented.

File: PatientWebApplication/HealthClinic.DAL/Utility/UtilityMethods.cs [hunk 3]
```diff
@@ -40,5 +40,20 @@ public static Boolean TryParseTimeSpanInCorrectFormat(String timeSpan)
             var time = new TimeSpan();
             return TimeSpan.TryParseExact(timeSpan, "hh:\\mm\\:ss", CultureInfo.InvariantCulture, out time);
         }
+
+        public static Boolean CheckIfDateIsToday(string dateString)
+        {
+            var date = ParseDateInCorrectFormat(dateString);
+            var today = DateTime.Now;
+            if (date.Date == today.Date && date.Month == today.Month && date.Year == today.Year) return true;
+            return false;
```
Comment (trki293) [inline PatientWebApplication/HealthClinic.DAL/Utility/UtilityMethods.cs:49]: Refact to write in one line of code.

File: PatientWebApplication/HealthClinic.DAL/Utility/UtilityMethods.cs [hunk 4]
```diff
@@ -40,5 +40,20 @@ public static Boolean TryParseTimeSpanInCorrectFormat(String timeSpan)
...
+        }
+
+        public static Boolean CheckIfDateIsToday(DateTime date)
+        {
+            var today = DateTime.Now;
+            if (date.Date == today.Date && date.Month == today.Month && date.Year == today.Year) return true;
+            return false;
```
Comment (trki293) [inline PatientWebApplication/HealthClinic.DAL/Utility/UtilityMethods.cs:56]: Refact to write in one line of code.

**Review by trki293 — CHANGES_REQUESTED** [review]
Code changes required.

**Review by trki293 — APPROVED** [review]
PR is approved. Good Clean code practice are followed.

**Pull request merged**